### PR TITLE
[SW-2297] Exclude Content of site/.doctrees from SW Distribution Archive

### PR DESCRIPTION
**Pull request by mn-mikke**
The options `useDoctreeCache` and `doctreeCacheDirectory ` of  [`sphinx-gradle-plugin`](https://trustin.github.io/sphinx-gradle-plugin/configuration.html) seem to not have any effect.

### CODE REVIEW

**Comment by jakubhava** [comment]
@mn-mikke Was there any problem with documentation?

File: doc/build.gradle [hunk 1]
```diff
@@ -95,7 +99,8 @@ createVersionFile.dependsOn cleanDoc
 sphinx.dependsOn createVersionFile
 clean.dependsOn cleanDoc
 substitute.dependsOn sphinx
-sourcesDocJar.dependsOn substitute
+deleteDocTrees.dependsOn substitute
+sourcesDocJar.dependsOn deleteDocTrees
```
Comment (jakubhava) [inline doc/build.gradle:103]: `sourcesDocJar` does not need to depend on anything as it creates the jar from the src dir where no files are generated

**Comment by mn-mikke** [comment]
> @mn-mikke Was is the motivation for this change please?

The motivation is to reduce the size of SW distribution archive.

**Comment by jakubhava** [comment]
Thanks! I also found the Slack discussion, good idea!

**Pull request merged**